examples_tests/50.RadixSort/main.cpp

```diff
@@ -0,0 +1,555 @@
+				auto cs_rawptr = cs.get();
+		
+				shader_gpu = driver->getGPUObjectsFromAssets(&cs_rawptr, &cs_rawptr + 1)->front();
+			}
+		
+			global_prefix_sum_pipeline = driver->createGPUComputePipeline(nullptr, std::move(pipeline_layout), std::move(shader_gpu));
+		}
+
+		// Dispatch for the final global scan array
+
+		{
+			const uint32_t count = 2;
+			IGPUDescriptorSet::SDescriptorInfo ds_info[count];
+			ds_info[0].desc = in;
+			ds_info[0].buffer = { 0u, in_size };
+
+			// Todo: You probably don't need it to update every iteration
+			ds_info[1].desc = partial_reductions_gpu;
+			ds_info[1].buffer = { 0u, partial_reductions_size };
+
+			IGPUDescriptorSet::SWriteDescriptorSet writes[count];
+			for (uint32_t i = 0; i < count; ++i)
+				writes[i] = { ds_global_prefix_sum.get(), i, 0u, 1u, asset::EDT_STORAGE_BUFFER, ds_info + i };
+
+			driver->updateDescriptorSets(count, writes, 0u, nullptr);
+
+			driver->bindComputePipeline(global_prefix_sum_pipeline.get());
+			driver->bindDescriptorSets(video::EPBP_COMPUTE, global_prefix_sum_pipeline->getLayout(), 0u, 1u, &ds_global_prefix_sum.get(), nullptr);
+
+			driver->pushConstants(global_prefix_sum_pipeline->getLayout(), asset::ISpecializedShader::ESS_COMPUTE, 0u, sizeof(uint32_t), &in_count);
+			driver->dispatch(wg_count, 1, 1);
+		}
+	}
+}
+
+int main()
+{
+	// create device with full flexibility over creation parameters
+	// you can add more parameters if desired, check nbl::SIrrlichtCreationParameters
+	nbl::SIrrlichtCreationParameters params;
+	params.Bits = 24; //may have to set to 32bit for some platforms
+	params.ZBufferBits = 24; //we'd like 32bit here
+	params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
+	params.WindowSize = dimension2d<uint32_t>(512, 512);
+	params.Fullscreen = false;
+	params.Vsync = true; //! If supported by target platform
+	params.Doublebuffer = true;
+	params.Stencilbuffer = false; //! This will not even be a choice soon
+	auto device = createDeviceEx(params);
+
+	if (!device)
+		return 1; // could not create selected driver.
+
+	IVideoDriver* driver = device->getVideoDriver();
+
+	io::IFileSystem* filesystem = device->getFileSystem();
+	asset::IAssetManager* am = device->getAssetManager();
+
+	GLint64 max_ssbo_size;
+	video::COpenGLExtensionHandler::extGlGetInteger64v(GL_MAX_SHADER_STORAGE_BLOCK_SIZE, &max_ssbo_size);
+
+	std::cout << "\nMax SSBO size: " << max_ssbo_size << " bytes" << std::endl;
+
+	GLint max_wg_count;
+	video::COpenGLExtensionHandler::extGlGetIntegeri_v(GL_MAX_COMPUTE_WORK_GROUP_COUNT, 0, &max_wg_count);
+
+	std::cout << "Max WG count: " << max_wg_count << std::endl;
+
+	GLint max_wg_dim;
+	video::COpenGLExtensionHandler::extGlGetIntegeri_v(GL_MAX_COMPUTE_WORK_GROUP_SIZE, 0, &max_wg_dim);
+	std::cout << "Max WG dim: " << max_wg_dim << std::endl;
+
+	const size_t in_count = 1 << 28; // > 64 MB, this param is tied to macros in Histogram.comp for now
```

Comment:
this is 256M elements, not 16M

Reply:
Trying to push my implementation, `1 << 23` is what I want